### PR TITLE
Rewrite list tests

### LINKED ISSUE #30: Eliminate no_network from magic_folder.test.cli.test_magic_folder.ListMagicFolder
Child of #22

### PR DESCRIPTION
**Pull request by exarkun**
Fixes #30

### CODE REVIEW

**Comment by exarkun** [comment]
Did this actually end up as a win?  It's certainly a lot more lines. :(  But the line count change misses the fact that these tests depend on a lot less code from Tahoe-LAFS now...

I really wanted to avoid having any subprocesses at all here but it's not possible to *really* create a new magic folder without Tahoe-LAFS storage capabilities.  The tests could directly mash `magic_folder.yaml` and perhaps test some things without a running Tahoe-LAFS node ... but I think that might go against the spirit of _these_ tests.  Perhaps there is room to exercise more cases in a different test suite that operates directly on `magic_folder.yaml` instead of a running Tahoe-LAFS node?

File: src/magic_folder/frontends/magic_folder.py [hunk 1]
```diff
@@ -328,10 +328,6 @@ def save_magic_folders(node_directory, folders):
         yamlutil.safe_dump({u"magic-folders": folders}),
     )
 
-    config = configutil.get_config(os.path.join(node_directory, u"tahoe.cfg"))
-    configutil.set_config(config, "magic_folder", "enabled", "True")
-    configutil.write_config(os.path.join(node_directory, u"tahoe.cfg"), config)
-
```
Comment (exarkun) [inline src/magic_folder/frontends/magic_folder.py]: The `magic-folder list` tests invoke `magic-folder create` and other mutate-y commands.  Those commands all want to upgrade and rewrite the magic-folder configuration.  Previously this made sense but now the `[magic_folder]enabled` setting in `tahoe.cfg` is worse than useless.  We don't want to enable Tahoe-LAFS' built-in magic-folders support if we're running the stand-alone magic-folder.  So, the stand-alone magic-folder should never rewrite the Tahoe-LAFS config to turn on this toggle.

File: src/magic_folder/scripts/magic_folder_cli.py [hunk 2]
```diff
@@ -3,8 +3,10 @@
 import os
 import sys
 import urllib
+from six.moves import (
+    StringIO as MixedIO,
+)
```
Comment (exarkun) [inline src/magic_folder/scripts/magic_folder_cli.py]: Should not be a behavior change but should be more clear to readers.  This is not a byte stream.  This is not a text stream.  This is a whatever-you-feel-like stream.  This is bad and should be fixed but the implementation makes no consistent decision about whether it wants to write out bytes or text  and I didn't want to fix every string literal in the whole project in this PR.

File: src/magic_folder/scripts/magic_folder_cli.py [hunk 3]
```diff
@@ -1059,7 +1061,7 @@ def do_magic_folder(options):
     try:
         return f(so)
     except Exception as e:
-        print("Error: %s" % (e,), file=options.stderr)
+        print(u"Error: {}".format(e), file=options.stderr)
```
Comment (exarkun) [inline src/magic_folder/scripts/magic_folder_cli.py]: I did get tricked into *starting* to change every string literal in the whole project before I realized that was a trap, though...

File: src/magic_folder/test/cli/test_magic_folder.py [hunk 4]
```diff
@@ -290,48 +316,162 @@ def get_Bob_magicfolder(result):
         return d
 
 
-class ListMagicFolder(MagicFolderCLITestMixin, AsyncTestCase):
-
+@defer.inlineCallbacks
+def create_introducer(testcase, introducer_directory):
+    """
+    Make an run a Tahoe-LAFS introducer node.
```
Comment (vu3rdd) [inline src/magic_folder/test/cli/test_magic_folder.py]: Typo here? "Make a Tahoe-LAFS introducer node? ?

**Pull request merged**